src/main/java/org/sagebionetworks/bridge/services/AuthenticationService.java

```diff
@@ -491,12 +496,38 @@ protected String generateReauthToken() {
         return SecureTokenGenerator.INSTANCE.nextToken();
     }
     
+    public UserSession oauthSignIn(CriteriaContext context, OAuthAuthorizationToken authToken) {
+        String synapseUserId = oauthProviderService.oauthSignIn(authToken);
+        
+        // This has not been observed to happen but in theory the user could deny access to their
+        // user ID, preventing this from being returned despite a successful OAuth exchange.
+        if (synapseUserId == null) {
+            throw new EntityNotFoundException(Account.class);
+        }
+        
+        AccountId accountId = AccountId.forSynapseUserId(authToken.getStudyId(), synapseUserId);
+        Account account = accountDao.getAccount(accountId);
+        if (account == null) {
+            throw new EntityNotFoundException(Account.class);
+        }
+        
+        clearSession(authToken.getStudyId(), account.getId());
+        Study study = studyService.getStudy(authToken.getStudyId());
+        UserSession session = getSessionFromAccount(study, context, account);
+
+        cacheProvider.setUserSession(session);
+        if (!session.doesConsent() && !session.isInRole(Roles.ADMINISTRATIVE_ROLES)) {
+            throw new ConsentRequiredException(session);
+        }
```

Comment:
Is the consent check necessary? We don't intend for participants to sign-in with Synapse. (In fact, we explicitly do _not_ want to do that.) Would it be better to check for Admin privileges and throw a 403 if they're missing?

Reply:
A much better plan. Changed this to check for roles and no more consent check.